src/circular-queue/iterator.ts

```diff
@@ -0,0 +1,25 @@
+import {ADTCircularQueue} from '../circular-queue';
+import {IterableType} from '../iterable-type';
+import {Iterator} from '../iterator';
+
+export class CircularQueueIterator<ItemT> implements Iterator<ItemT | null> {
+	private curr: number;
+	private circularQueue: ADTCircularQueue<ItemT>;
+
+	constructor(cq: ADTCircularQueue<ItemT>) {
+		this.circularQueue = cq;
+		this.curr = 0;
+	}
+
+	next(): IterableType<ItemT | null> {
+		if (this.circularQueue.isEmpty() || this.curr >= this.circularQueue.size()) {
+			return {
```

Comment:
I recommend creating an exported helper function in an external file which takes two arguments and returns the `IterableType<T>` object to guarantee consistency and reduce code duplication across all the ADT implementations.

Additionally, in the future if we need to add a property to the type, we would add an extra function parameter which would immediately cause a build error for all ADT functions which don't provide all required parameters, which is good.

The helper would look something like
```typescript
export function makeIterableType<T>(value: T, done: boolean): IterableType<T> {
   return {
      value: value,
      done: typeof done === 'boolean' ? done : true
   };
}
```
Then, in each place it's used just call:
```typescript
return makeIterableType<ItemT | null>(null, true);
```

- [ ] Add `makeIterableType` helper and replace Iterable Type object returns with the helper return.
- [ ] Write tests to cover helper functionality. It shouldn't take very much.